tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/Emitter.scala

```diff
@@ -198,16 +198,32 @@ final class Emitter private (semantics: Semantics, outputMode: OutputMode,
         }
 
       case ModuleKind.ESModule =>
-        foreachImportedModule { (module, pos0) =>
+        foreachImportedModule { (module, pos0, path) =>
           implicit val pos = pos0
           val from = js.StringLiteral(module)
           val moduleBinding = jsGen.envModuleField(module).ident
-          val importStat = js.ImportNamespace(moduleBinding, from)
-          builder.addJSTree(importStat)
+          path match {
+            // namespace import or default import
+            case Nil | List("default") =>
```

Comment:
Namespace and default imports are not the same thing. A default import is actually much more similar to a named import, whose name happens to be `default`. You cannot treat a default import as a namespace import.